app/apollo/resolvers/common.js

```diff
@@ -55,53 +55,76 @@ const validClusterAuth = async (me, queryName, context) => {
   }
 };
 
-var getAllowedChannels = async(me, org_id, action, field, context)=>{
+// Polymorphically check for cached IAM decision, Get resources authorized by Access Policy, Update cache for individual resource authentication
+var getAllowedResources = async(me, org_id, action, type, queryName, context, searchByTags=null, searchByUuid=null, searchQuery=null)=>{
   const { models } = context;
-  var channels = await models.Channel.find({ org_id });
-  return await filterChannelsToAllowed(me, org_id, action, field, channels, context);
-};
 
-var filterChannelsToAllowed = async(me, org_id, action, field, channels, context)=>{
-  const { models } = context;
-  var decisionInputs = _.map(channels, (channel)=>{
-    return {
-      type: field,
-      action,
-      uuid: channel.uuid,
-      name: channel.name,
-    };
-  });
-  var decisions = await models.User.isAuthorizedBatch(me, org_id, decisionInputs, context);
-  channels = _.filter(channels, (val, idx)=>{
-    return decisions[idx];
-  });
-  return channels;
-};
+  var resources;
 
-var getAllowedSubscriptions = async(me, org_id, action, field, context)=>{
-  const { models } = context;
-  var subscriptions = await models.Subscription.find({ org_id });
-  return await filterSubscriptionsToAllowed(me, org_id, action, field, subscriptions, context);
+  // Check for cached IAM decision, returns true if cache is empty or all is authorized
```

Comment:
```suggestion
  // Check for cached IAM decision, returns true if **cache misses** or **cache confirms user is authorized on all resources of the specified type**
```
- Clarifies that it deals with cache misses, not "cache is empty"
- Clarifies what "all is authorized" means
- **NOTE**: As noted in at least one other review comment (cluster.js), the design of this function appears to be incorrect however -- as described, it's impossible to tell the difference between "cache says user **is** authorized on all" and "cache doesn't know", yet is used to determine whether fine-grained-auth checks can later be skipped.  Fine grained auth checks can only be skipped if user **is** authorized on all, not on cache miss.